### PR TITLE
ADEN-5773 Mobile tests for CTP Fan Takeover 

### PR DESCRIPTION
**Pull request by nandy-andy**

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapMercury.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    vuap.clickOnArea(2);
+
+    List<String> tabs = new ArrayList<String>(driver.getWindowHandles());
+    Assert.assertTrue(
+        tabs.size() == 3,
+        "There should be three tabs opened after two clicks."
+    );
+    final String actual = ads.switchToNewBrowserTab();
+    Assert.assertTrue(
+        actual.equals("http://fandom.wikia.com/"),
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapMercury.java]: MAJOR Move the "http://fandom.wikia.com/" string literal on the left side of this string comparison. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1132)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 2 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AdsBaseObject.java#L409](https://github.com/Wikia/selenium-tests/blob/fd020ae653e530c1448ad0d1ab13a3c3dd608d13/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L409): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java [hunk 2]
```diff
@@ -721,6 +721,18 @@ public void scrollToPosition(String selector) {
...
   }
 
+  public void scrollToSlot(String slotName) {
+    if (slotName.equals(AdsContent.BOTTOM_LB)) {
+      triggerComments();
+    }
+
+    if (slotName.equals(AdsContent.MOBILE_BOTTOM_LB)) {
+      scrollToFooter();
+    }
```
Comment (dianafa) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java]: nitpick: `if ... else` would be more appropriate here

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapMercury.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    ads.scrollToSlot(slot);
+
+    vuap.clickOnArea(1);
+    vuap.clickOnArea(2);
+
+    List<String> tabs = new ArrayList<String>(driver.getWindowHandles());
+    Assert.assertTrue(
+        tabs.size() == 3,
+        "There should be three tabs opened after two clicks."
+    );
```
Comment (dianafa) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapMercury.java]: as discussed recently, the assertion
```
Assertion.assertEquals(tabs.size(), 3, "There should be three tabs opened after two clicks.");
```
is much more readable and easier to debug what actually went wrong

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapMercury.java (hunk 1 above)
Comment (dianafa) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapMercury.java]: `Assertion.assertEquals(actual, "http://fandom.wikia.com/", "Top part of creative should point to FANDOM page but it points to " + actual` ?

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AdsBaseObject.java#L409](https://github.com/Wikia/selenium-tests/blob/7dbc85c6b469d2221781c4a7d61b51250ec822e1/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L409): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AdsBaseObject.java#L409](https://github.com/Wikia/selenium-tests/blob/31852c3acb95843752cad100674905f94383ef24/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L409): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Pull request merged**